### PR TITLE
#2116 Add missing menus "Validate Model" on SemanticBrowser

### LINKED ISSUE #2116: Add missing menus on Semantic Browser and Interpreter
1. delete menu in SB: The menu shall be enabled on Elements and
DRepresentationDescriptor.
2. add missing menus in SB: validate model, rec/repl, transitions, wizzards,
allocation management, modelling accelerators.
3. group Copy As menus, add it also in SB
4. dynamic group.new, group.open, add it in SB

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Resolves #2116

Change-Id: I5d98e85bae8dad3ea4e5d0766997794e6ecb8323
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.validation.ui/plugin.xml [hunk 1]
```diff
@@ -68,8 +56,7 @@
    
    
    <extension point="org.eclipse.ui.menus">
-      <menuContribution allPopups="false" locationURI="popup:org.eclipse.gmf.runtime.diagram.ui.DiagramEditorContextMenu?before=org.polarsys.capella.common.re">
-
+      <menuContribution locationURI="popup:org.eclipse.ui.popup.any?before=group.rec">
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.validation.ui/plugin.xml:59]: after=group.validation no ?

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java [hunk 2]
```diff
@@ -58,4 +58,24 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
...
     return true;
   }
+  
+  private List<EObject> getEObjectsToRemove(List<?> elementsToRemove) {
+    List<EObject> eObjects = elementsToRemove.stream()//
+        .filter(EObject.class::isInstance) //
+        .map(EObject.class::cast).collect(Collectors.toList());
+    
+    List<EObject> eObjectsToRemove = new ArrayList<EObject>();
+    for (EObject elem : eObjects) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java:68]: I think its what does CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)

**Review by pdulth — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java [hunk 3]
```diff
@@ -36,10 +39,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
...
     if (selection instanceof IStructuredSelection) {
       List<?> elementsToRemove = ((IStructuredSelection) selection).toList();
-      List<EObject> eObjectsToRemove = elementsToRemove.stream()//
-          .filter(EObject.class::isInstance) //
-          .map(EObject.class::cast).collect(Collectors.toList());
-
+      List<EObject> eObjectsToRemove = getEObjectsToRemove(elementsToRemove);
       if (!eObjectsToRemove.isEmpty()) {
 
         ExecutionManager executionManager = TransactionHelper.getExecutionManager(eObjectsToRemove);
```
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java:43]: **It should be eObjectsToRemove!**

 new CapellaDeleteCommand(executionManager, ~~elementsToRemove~~ eObjectsToRemove, true,
withConfirmDeletion(), true)`

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java (hunk 2 above)
Comment (sandupostaru) [reply core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java]: It should be `CapellaAdapterHelper.resolveBusinessObject`, we can not use `CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)` since we want to keep `DRepresentationDescriptor` as they are, and not obtain their target (I guess if the user clicks on a descriptor we want to delete the descriptor, and not the semantic target).

File: core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java [hunk 4]
```diff
@@ -114,10 +115,11 @@ public ValidateAction createDefaultValidation() {
    * @return
    */
   protected IContributionItem createContributionItem() {
-
     if (userValidationActions.size() > 0) {
       IMenuManager menuManager = new MenuManager("Validate Model", imageDescriptor, "ID"); //$NON-NLS-1$ //$NON-NLS-2$
-      menuManager.add(defaultValidationAction);
+      if (defaultValidationAction.isEnabled()) {
```
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java:120]: What is the reason behind this change?

**Review by sandupostaru — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.validation.ui/plugin.xml (hunk 1 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.validation.ui/plugin.xml]: done.

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java [hunk 5]
```diff
@@ -36,10 +37,7 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
...
     if (selection instanceof IStructuredSelection) {
       List<?> elementsToRemove = ((IStructuredSelection) selection).toList();
-      List<EObject> eObjectsToRemove = elementsToRemove.stream()//
-          .filter(EObject.class::isInstance) //
-          .map(EObject.class::cast).collect(Collectors.toList());
-
+      Collection<EObject> eObjectsToRemove = CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(elementsToRemove);
       if (!eObjectsToRemove.isEmpty()) {
 
         ExecutionManager executionManager = TransactionHelper.getExecutionManager(eObjectsToRemove);
```
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java:43]: It should be `CapellaAdapterHelper.resolveBusinessObject`, we can not use `CapellaAdapterHelper.resolveDescriptorsOrBusinessObjects(eObjects)` since we want to keep `DRepresentationDescriptor` as they are, and not obtain their target (I guess if the user clicks on a descriptor we want to delete the descriptor, and not the semantic target).

**Review by sandupostaru — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java (hunk 5 above)
Comment (sandupostaru) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/handlers/DeleteHandler.java:43]: CapellaDeleteCommand deleteCommand = new CapellaDeleteCommand(executionManager, ~~elementsToRemove~~ eObjectsToRemove, true,
withConfirmDeletion(), true);

**Review by sandupostaru — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java (hunk 4 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.validation.ui/src/org/polarsys/capella/core/validation/ui/DynamicActionProvider.java:120]: The action si grayed out if disable, a menu si created but does nothing.

**Review by sandupostaru — APPROVED** [review]

**Review by pdulth — APPROVED** [review]
